cadc-dali/src/main/java/ca/nrc/cadc/dali/tables/parquet/ParquetWriter.java

```diff
@@ -0,0 +1,205 @@
+package ca.nrc.cadc.dali.tables.parquet;
+
+import ca.nrc.cadc.dali.tables.TableData;
+import ca.nrc.cadc.dali.tables.TableWriter;
+import ca.nrc.cadc.dali.tables.votable.VOTableDocument;
+import ca.nrc.cadc.dali.tables.votable.VOTableResource;
+import ca.nrc.cadc.dali.tables.votable.VOTableWriter;
+import ca.nrc.cadc.dali.util.FormatFactory;
+
+import java.io.IOException;
+import java.io.OutputStream;
+import java.io.StringWriter;
+import java.io.Writer;
+
+import java.util.HashMap;
+import java.util.Iterator;
+import java.util.List;
+import java.util.Map;
+
+import org.apache.avro.Schema;
+import org.apache.avro.generic.GenericData;
+import org.apache.avro.generic.GenericRecord;
+import org.apache.hadoop.conf.Configuration;
+import org.apache.log4j.Logger;
+import org.apache.parquet.avro.AvroParquetWriter;
+import org.apache.parquet.hadoop.ParquetFileWriter;
+import org.apache.parquet.hadoop.metadata.CompressionCodecName;
+import org.apache.parquet.io.OutputFile;
+import org.apache.parquet.io.PositionOutputStream;
+
+public class ParquetWriter implements TableWriter<VOTableDocument> {
+
+    private static final Logger log = Logger.getLogger(ParquetWriter.class);
+
+    private FormatFactory formatFactory;
+
+    public ParquetWriter() {
+
+    }
+
+    @Override
+    public String getExtension() {
+        return "parquet";
+    }
+
+    @Override
+    public String getContentType() {
+        return "application/vnd.apache.parquet";
+    }
+
+    @Override
+    public String getErrorContentType() {
+        return "text/plain";
+    }
+
+    @Override
+    public void setFormatFactory(FormatFactory ff) {
+        this.formatFactory = ff;
+    }
+
+    @Override
+    public void write(VOTableDocument voTableDocument, OutputStream out) throws IOException {
+        write(voTableDocument, out, Long.MAX_VALUE);
+    }
+
+    @Override
+    public void write(VOTableDocument voTableDocument, OutputStream out, Long maxRec) throws IOException {
+        log.debug("ParquetWriter Write service called. MaxRec = " + maxRec);
+        for (VOTableResource resource : voTableDocument.getResources()) {
+            Schema schema = DynamicSchemaGenerator.generateSchema(resource.getTable().getFields());
+            OutputFile outputFile = outputFileFromStream(out);
+
+            TableData tableData = resource.getTable().getTableData();
+            resource.getTable().setTableData(null);
+
+            StringWriter stringWriter = new StringWriter();
+            VOTableWriter votableWriter = new VOTableWriter();
+            votableWriter.write(voTableDocument, stringWriter, maxRec);
+
+            Map<String, String> customMetaData = new HashMap<>();
+            customMetaData.put("votable", stringWriter.toString());
```

Comment:
The note on VOTable in parquet specifies two specific keys to use here.

https://github.com/ivoa/voparquet